common/txmgr/inmemory_store.go

```diff
@@ -0,0 +1,1869 @@
+	if ms.chainID.String() != chainID.String() {
+		return nil, fmt.Errorf("find_txes_pending_callback: %w", ErrInvalidChainID)
+	}
+
+	filterFn := func(tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) bool {
+		if tx.TxAttempts == nil || len(tx.TxAttempts) == 0 {
+			return false
+		}
+
+		if tx.TxAttempts[0].Receipts == nil || len(tx.TxAttempts[0].Receipts) == 0 {
+			return false
+		}
+
+		if tx.PipelineTaskRunID.Valid && tx.SignalCallback && !tx.CallbackCompleted &&
+			tx.TxAttempts[0].Receipts[0].GetBlockNumber() != nil &&
+			big.NewInt(blockNum-int64(tx.MinConfirmations.Uint32)).Cmp(tx.TxAttempts[0].Receipts[0].GetBlockNumber()) > 0 {
+			return true
+		}
+
+		return false
+
+	}
+	states := []txmgrtypes.TxState{TxConfirmed}
+	txs := []txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+	for _, as := range ms.addressStates {
+		txs = append(txs, as.FetchTxs(states, filterFn)...)
+	}
+
+	receiptsPlus := make([]txmgrtypes.ReceiptPlus[R], len(txs))
+	meta := map[string]interface{}{}
+	for i, tx := range txs {
+		if err := json.Unmarshal(json.RawMessage(*tx.Meta), &meta); err != nil {
+			return nil, err
+		}
+		failOnRevert := false
+		if v, ok := meta["FailOnRevert"].(bool); ok {
+			failOnRevert = v
+		}
+
+		receiptsPlus[i] = txmgrtypes.ReceiptPlus[R]{
+			ID:           tx.PipelineTaskRunID.UUID,
+			Receipt:      (tx.TxAttempts[0].Receipts[0]).(R),
+			FailOnRevert: failOnRevert,
+		}
+		clear(meta)
+	}
+
+	return receiptsPlus, nil
+}
+
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) UpdateTxCallbackCompleted(ctx context.Context, pipelineTaskRunRid uuid.UUID, chainId CHAIN_ID) error {
+	if ms.chainID.String() != chainId.String() {
+		return fmt.Errorf("update_tx_callback_completed: %w", ErrInvalidChainID)
+	}
+
+	// Persist to persistent storage
+	if err := ms.txStore.UpdateTxCallbackCompleted(ctx, pipelineTaskRunRid, chainId); err != nil {
+		return err
+	}
+
+	// Update in memory store
+	fn := func(tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) {
+		if tx.PipelineTaskRunID.UUID == pipelineTaskRunRid {
+			tx.CallbackCompleted = true
+		}
+	}
+	wg := sync.WaitGroup{}
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+	for _, as := range ms.addressStates {
+		wg.Add(1)
+		go func(as *AddressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) {
+			as.ApplyToTxsByState(nil, fn)
+			wg.Done()
+		}(as)
+	}
+	wg.Wait()
+
+	return nil
+}
+
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) SaveFetchedReceipts(ctx context.Context, receipts []R, chainID CHAIN_ID) error {
+	if ms.chainID.String() != chainID.String() {
+		return fmt.Errorf("save_fetched_receipts: %w", ErrInvalidChainID)
+	}
+
+	// Persist to persistent storage
+	if err := ms.txStore.SaveFetchedReceipts(ctx, receipts, chainID); err != nil {
+		return err
+	}
+
+	// Update in memory store
```

Comment:
Are these changes to in-memory store same as the persistent store?
In the EvmTxStore, I see many things being done for this.
We update fields in attempt, we change Tx state,